dashboard/src/app/workspaces/workspace-details/environments/environments.controller.ts

```diff
@@ -54,23 +64,30 @@ export class WorkspaceEnvironmentsController {
     $scope.$watch(() => {
       return this.workspaceConfig.environments;
     }, () => {
-      this.init();
+      if (Object.keys(this.workspaceConfig.environments).length > 0) {
+        this.init();
+      }
     });
   }
 
   /**
    * Sets initial values
    */
-  init() {
+  init(): void {
     this.newEnvironmentName = this.environmentName;
     this.environment = this.workspaceConfig.environments[this.environmentName];
 
+    if (!this.environment.recipe) {
+      return;
```

Comment:
This is about stack's selector initialization. I will move this check into suitable place.